clang/test/SemaCXX/cxx2c-binding-pack.cpp

```diff
@@ -188,3 +188,19 @@ void other_main() {
   static_assert(f<int>() == 2);
 }
 }  // namespace
+
+
+namespace GH125165 {
+
+template <typename = void>
+auto f(auto t) {
+    const auto& [...pack] = t;
+    // expected-error@-1 {{cannot decompose non-class, non-array type 'char const'}}
+    (pack, ...);
+};
+
+void g() {
+    f('x'); // expected-note {{in instantiation}}
```

Comment:
Interesting, should this work: https://godbolt.org/z/dKcaa3jY4

We should also try other arguments types beside *char* like *int*, *float* etc

Reply:
That happens to work because the pack can be empty. In clang-19 it seems to still allow decomposition. https://godbolt.org/z/zMrqTYE8c

Reply:
Maybe: https://godbolt.org/z/Ejf8Mj5jj